src/rshim_pcie.c

```diff
@@ -270,8 +278,13 @@ rshim_pcie_read(rshim_backend_t *bd, int chan, int addr, uint64_t *result)
   dev->write_count = 0;
 
 #ifndef __LP64__
+  if (rshim_is_bluefield3(dev->pci_dev->device_id))
+    return -EIO;
+
   rc = rshim_byte_acc_read(dev, RSH_CHANNEL_BASE(chan) + addr, result);
 #else
+  if (rshim_is_bluefield3(dev->pci_dev->device_id))
+    chan = 0;
   *result = readq(dev->rshim_regs + (addr | (chan << 16)));
```

Comment:
I think we might need a channel map here for PCIE like below.  Or else it can only access the first 64K address with this BAR mapping.

if ((rshim_is_bluefield3(dev->pci_dev->device_id)) {
    *result = readq(dev->rshim_regs + bf3_rshim_pcie_chan_map[chan])
} else {
    *result = readq(dev->rshim_regs + (addr | (chan << 16)));
}

The bf3_rshim_channel_map will have some definition like below, similar to bf3_wvalue_widx_pair_map with adjusted offset.

uint32_t bf3_rshim_pcie_chan_map[] = {
    [RSHIM_CHANNEL] = 0;
    [UART0_CHANNEL] = ...;
};

Reply:
Do you think that would be useful given that we will only be allowed to access the registers with "BOOT" privilege in this path?

Reply:
I see. Is it the mapping we added in BL1?

Reply:
Yes, we program the PRNF_DMA_CRWIN in BL1 to map this.